### PR TITLE
Add traffic splitting example.

### PR DESCRIPTION
**Pull request by markusthoemmes**
TODO:

1. ~~Wait for a new version of KOBW to support tags.~~
2. ~~Push the tags to the "official" repository.~~
3. ~~Adjust the description of the initial service a bit to include the version.~~

### CODE REVIEW

File: knative-kubecon/build/010-build-template.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
     default: /workspace
   - name: OC_BUILDER_IMAGE
     description: The name of the builder image to use
-    default: docker.io/vdemeester/kobw-builder:0.1.0
+    default: docker.io/vdemeester/kobw-builder:0.1.1
```
Comment (vdemeester) [inline knative-kubecon/build/010-build-template.yaml:21]: the template in the `README` should be updated too :angel:

**Review by vdemeester — CHANGES_REQUESTED** [review]

File: knative-kubecon/README.md [hunk 2]
```diff
@@ -387,7 +390,69 @@ the *testevents* pod we've created earlier, which is the Source listening on Kub
 We can also roughly spot how the Knative Serving system works underneath, as our deployment gets requests from the activator and is connected
 to the autoscaler, to provide metrics there.
 
-## 4. Scaling up and down
+What we also spot: Our application has a problem. All requests from the dispatcher to the application are failing.
```
Comment (matzew) [inline knative-kubecon/README.md:393]: ehm, I don't really get this, looking at the last(above) image from 3.

See: https://github.com/openshift-cloud-functions/demos/blob/ca996669772dd79d9b2f141d56f7ce4529e782b8/knative-kubecon/images/kiali.png

IMO the traffic is green, from dispatcher to the app -

File: knative-kubecon/README.md [hunk 3]
```diff
@@ -387,7 +390,69 @@ the *testevents* pod we've created earlier, which is the Source listening on Kub
...
+>     rolloutPercent: 50
+```
+
+Apply these changes via:
+
+```bash
+oc apply -f serving/012-service-traffic.yaml
+```
+
+The system will now evenly divide the traffic between the two deployed versions. Using Kiali, we can verify that the traffic to the new version is indeed
```
Comment (matzew) [inline knative-kubecon/README.md:446]: should we do a new rollout to 100% ?

File: knative-kubecon/README.md [hunk 4]
```diff
@@ -387,7 +390,86 @@ the *testevents* pod we've created earlier, which is the Source listening on Kub
...
+9c9
+<     rolloutPercent: 50
+---
+>     rolloutPercent: 100
+```
+
+Apply these changes via:
+
+```bash
+oc apply -f serving/013-service-final.yaml
```
Comment (matzew) [inline knative-kubecon/README.md:472]:
```
➜  knative-kubecon git:(2bfdf9c) oc apply -f serving/013-service-final.yaml 
Error from server (InternalError): error when applying patch:
{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"serving.knative.dev/v1alpha1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"helloworld-openshift\",\"namespace\":\"myproject\"},\"spec\":{\"release\":{\"configuration\":{\"build\":{\"source\":{\"git\":{\"revision\":\"v2\",\"url\":\"https://github.com/openshift-cloud-functions/openshift-knative-application\"}},\"template\":{\"arguments\":[{\"name\":\"IMAGE_STREAM\",\"value\":\"golang:1.11\"},{\"name\":\"IMAGE\",\"value\":\"helloworld:latest\"},{\"name\":\"NAME\",\"value\":\"helloworld-build\"}],\"name\":\"openshift-builds\"}},\"revisionTemplate\":{\"metadata\":{\"annotations\":{\"alpha.image.policy.openshift.io/resolve-names\":\"*\"}},\"spec\":{\"container\":{\"image\":\"docker-registry.default.svc:5000/myproject/helloworld:latest\",\"imagePullPolicy\":\"Always\"},\"containerConcurrency\":3}}},\"revisions\":[\"helloworld-openshift-00001\",\"helloworld-openshift-00002\"],\"rolloutPercent\":100}}}\n"}},"spec":{"release":{"rolloutPercent":100}}}
to:
Resource: "serving.knative.dev/v1alpha1, Resource=services", GroupVersionKind: "serving.knative.dev/v1alpha1, Kind=Service"
Name: "helloworld-openshift", Namespace: "myproject"
Object: &{map["apiVersion":"serving.knative.dev/v1alpha1" "kind":"Service" "metadata":map["namespace":"myproject" "resourceVersion":"12683" "selfLink":"/apis/serving.knative.dev/v1alpha1/namespaces/myproject/services/helloworld-openshift" "uid":"ad404b88-f491-11e8-8706-525400e8ab63" "annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"serving.knative.dev/v1alpha1\",\"kind\":\"Service\",\"metadata\":{\"annotations\":{},\"name\":\"helloworld-openshift\",\"namespace\":\"myproject\"},\"spec\":{\"release\":{\"configuration\":{\"build\":{\"source\":{\"git\":{\"revision\":\"v2\",\"url\":\"https://github.com/openshift-cloud-functions/openshift-knative-application\"}},\"template\":{\"arguments\":[{\"name\":\"IMAGE_STREAM\",\"value\":\"golang:1.11\"},{\"name\":\"IMAGE\",\"value\":\"helloworld:latest\"},{\"name\":\"NAME\",\"value\":\"helloworld-build\"}],\"name\":\"openshift-builds\"}},\"revisionTemplate\":{\"metadata\":{\"annotations\":{\"alpha.image.policy.openshift.io/resolve-names\":\"*\"}},\"spec\":{\"container\":{\"image\":\"docker-registry.default.svc:5000/myproject/helloworld:latest\",\"imagePullPolicy\":\"Always\"},\"containerConcurrency\":3}}},\"revisions\":[\"helloworld-openshift-00001\",\"helloworld-openshift-00002\"],\"rolloutPercent\":50}}}\n"] "creationTimestamp":"2018-11-30T11:18:34Z" "generation":'\x01' "name":"helloworld-openshift"] "spec":map["generation":'\x03' "release":map["configuration":map["build":map["source":map["git":map["revision":"v2" "url":"https://github.com/openshift-cloud-functions/openshift-knative-application"]] "template":map["arguments":[map["name":"IMAGE_STREAM" "value":"golang:1.11"] map["name":"IMAGE" "value":"helloworld:latest"] map["name":"NAME" "value":"helloworld-build"]] "name":"openshift-builds"]] "revisionTemplate":map["metadata":map["annotations":map["alpha.image.policy.openshift.io/resolve-names":"*"] "creationTimestamp":<nil>] "spec":map["container":map["image":"docker-registry.default.svc:5000/myproject/helloworld:latest" "imagePullPolicy":"Always" "name":"" "resources":map[]] "containerConcurrency":'\x03']]] "revisions":["helloworld-openshift-00001" "helloworld-openshift-00002"] "rolloutPercent":'2']] "status":map["latestCreatedRevisionName":"helloworld-openshift-00002" "latestReadyRevisionName":"helloworld-openshift-00002" "observedGeneration":'\x03' "traffic":[map["name":"current" "percent":'2' "revisionName":"helloworld-openshift-00001"] map["name":"candidate" "percent":'2' "revisionName":"helloworld-openshift-00002"] map["revisionName":"helloworld-openshift-00002" "name":"latest" "percent":'\x00']] "address":map["hostname":"helloworld-openshift.myproject.svc.cluster.local"] "conditions":[map["lastTransitionTime":"2018-11-30T11:27:46Z" "status":"True" "type":"ConfigurationsReady"] map["lastTransitionTime":"2018-11-30T11:27:46Z" "status":"True" "type":"Ready"] map["lastTransitionTime":"2018-11-30T11:27:46Z" "status":"True" "type":"RoutesReady"]] "domain":"helloworld-openshift.myproject.example.com" "domainInternal":"helloworld-openshift.myproject.svc.cluster.local"]]}
for: "serving/013-service-final.yaml": Internal error occurred: admission webhook "webhook.serving.knative.dev" denied the request: mutation failed: invalid value "100": spec.release.rolloutPercent
```

**Review by vdemeester — APPROVED** [review]
LGTM :boar:

**Pull request merged**